### PR TITLE
Support for ruby 3

### PR DESCRIPTION
**Pull request by kadru**
Hi

This is my intent to fix ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: on) for ruby 3 reported in  #65,  also fixes the failing tests for rails 4.2.

In my local two tests fails: `spec/fx/adapters/postgres/triggers_spec.rb:7` and `spec/fx/schema_dumper/trigger_spec.rb:4`, but I hope is because postgres version, I have postgres 12 installed.

### CODE REVIEW

File: lib/fx/statements/function.rb [hunk 1]
```diff
@@ -29,7 +29,10 @@ module Function
       #     $$ LANGUAGE plpgsql;
       #   SQL
       #
-      def create_function(name, version: 1, sql_definition: nil)
```
Comment (teoljungberg) [inline lib/fx/statements/function.rb:32]: I wonder if it's enough to use https://github.com/ruby/ruby2_keywords instead?

File: lib/fx/statements/trigger.rb [hunk 2]
```diff
@@ -98,7 +103,12 @@ def drop_trigger(name, on:, revert_to_version: nil)
       #         EXECUTE PROCEDURE uppercase_users_name();
       #    SQL
       #
-      def update_trigger(name, version: nil, on: nil, sql_definition: nil, revert_to_version: nil)
+      # def update_trigger(name, version: nil, on: nil, sql_definition: nil, revert_to_version: nil)
```
Comment (teoljungberg) [inline lib/fx/statements/trigger.rb:106]: This seems accidental :)

File: lib/fx/statements/function.rb (hunk 1 above)
Comment (kadru) [reply lib/fx/statements/function.rb:32]: Seems that doesn't  not fixes the errors, also print warnings: 
```
/home/oscar/workspace/fx/lib/fx/statements/function.rb:32: warning: Skipping set of ruby2_keywords flag for create_function (method accepts keywords or method does not accept argument splat)
```
I have used it this way:
```ruby
ruby2_keywords def create_function(name, version: 1, sql_definition: nil)
# rest of code...
```

File: lib/fx/statements/trigger.rb (hunk 2 above)
Comment (kadru) [reply lib/fx/statements/trigger.rb:106]: Sorry I wil remove that.

**Comment by pjungwir** [comment]
Thank you for this PR! I did pretty much the same thing in #76 except it's based on our latest changes and includes updates to the Travis & Appraisal files.

**Comment by teoljungberg** [comment]
Closing in favor of #76 .

**Pull request closed without merging**